lib/evmone_precompiles/bls.cpp

```diff
@@ -0,0 +1,42 @@
+#include "bls.hpp"
+#include <blst.hpp>
+
+namespace evmone::crypto::bls
+{
+bool g1_add(uint8_t* _rx, uint8_t* _ry, const uint8_t _x0[48], const uint8_t _y0[48],
+    const uint8_t _x1[48], const uint8_t _y1[48])
+{
+    blst::blst_fp x0;
+    blst::blst_fp y0;
+    blst::blst_fp x1;
+    blst::blst_fp y1;
+    blst::blst_fp_from_bendian(&x0, _x0);
```

Comment:
You'll want to check that these are less than the `fp` modulus.

Reply:
It was checked but on a different level. Now this check is moved to `g1_add` body